### ISSUE TITLE
[Inadyn] No automatic update according to period attribute

### ISSUE THREAD

**Issue by X1pheR** [alexbelgium/hassio-addons#154]
**Which addon?**
Inadyn

**Describe the bug**
The setting attribute "period" seems to not being handled. The addon output stays the same as displayed. I've let it run for a couple of days. Also as a note, the attributes "ttl" and "proxied" are not recognized as valid attributes (stated in ha logs) while they are valid according to inadyn documentation.

**To Reproduce**
1. Use configuration as in screenshots with real data.
2. Output will stay as in screenshots. 

**Expected behavior**
I would assume every 300 seconds I see some kind of update.

**Screenshots**
***Addon configuration***
```
providers:
  - provider: cloudflare.com
    ttl: 1
    proxied: true
    username: domain.eu
    password: my-personal-string
    hostname: home.domain.eu
period: 300
```
***Addon log output***
```
Starting...
/scripts/00-aaa_dockerfile_backup.sh: executing
/scripts/00-banner.sh: executing
-----------------------------------------------------------
 Add-on: Inadyn
 Inadyn is a small and simple Dynamic DNS, DDNS, client with HTTPS support. A large number of dynamic dns providers are supported (https://github.com/troglobit/inadyn#supported-providers). For those that are not, you can use a custom provider as per this project's documentation (https://github.com/nalipaz/hassio-addons/blob/master/README.md)
-----------------------------------------------------------
 Add-on version: 2.9.1
 You are running the latest version of this add-on.
 System: Home Assistant OS 7.1  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2021.12.8
 Home Assistant Supervisor: 2021.12.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums
 https://github.com/alexbelgium/hassio-addons
-----------------------------------------------------------
/scripts/99-run.sh: executing
inadyn[102]: In-a-dyn version 2.9.1 -- Dynamic DNS update client.
inadyn[102]: Guessing DDNS plugin 'default@cloudflare.com' from 'cloudflare.com'
inadyn[102]: Update forced for alias home.domain.eu, new IP# 84.xxx.xxx.xxx
inadyn[102]: Updating cache for home.hypershell.eu
```

**System**
* Supervisor version: 2021.12.2
* Host system version: 7.1 production

**Comment by alexbelgium** [comment]
Hi thanks I'll look at it. I don't use it so it will take a little time to check that. Thanks

**Comment by alexbelgium** [comment]
Hi, @deku-m, I remember you were the original requester of this addon. Do you still use it? Do you have any issues with updates? It could help me pinpoint the source of the issue. Thanks!

**Comment by X1pheR** [comment]
Thanks! if you want I could provide details to test with through a DM if that helps

**Comment by alexbelgium** [comment]
Thanks, to begin with I'll review tomorrow how the addon is structured :) 

For the moment, I've checked the inadyn issue log (https://github.com/troglobit/inadyn/issues) and there are some occurrences of non update, but they say it has been solved in the latest version. It's sometimes tricky to differentiate if the issue is due to the addon or the app itself if there are no specific error messages

**Comment by deku-m** [comment]
@alexbelgium  i don't use the period to update i think by default it updates every 12/24 hours or a restart.
Use it with strato.

**Comment by alexbelgium** [comment]
What do you mean, have you made an automation top at restarts the addon every x hours? That's what I've done to update my letsencrypt ssl files

**Comment by deku-m** [comment]
No i mean when a update is of home assistant or update from anything else that needs full reboot which also restarts the containers.

**Comment by alexbelgium** [comment]
Thanks! Although @X1pheR that can be a mitigation until it is fixed : to create an automation that restarts the addon every x hours. This works pretty well over already done it for another addon

**Comment by deku-m** [comment]
Could be that the issue is more likely in inadyn application then in the docker image here.
That for example period command isnt excepted.

**Comment by alexbelgium** [comment]
I think it is enabled in the new version I've pushed. Could you please try it? the period value is in seconds. The default in Inadyn is 30days, reason why you didn't see a change. Thanks!

**Comment by deku-m** [comment]
Not sure if this is the right output for the change ?
```
Starting...
/scripts/00-aaa_dockerfile_backup.sh: executing
/scripts/00-banner.sh: executing
hello
-----------------------------------------------------------
 Add-on: Inadyn
 Inadyn is a small and simple Dynamic DNS, DDNS, client with HTTPS support. A large number of dynamic dns providers are supported (https://github.com/troglobit/inadyn#supported-providers). For those that are not, you can use a custom provider as per this project's documentation (https://github.com/nalipaz/hassio-addons/blob/master/README.md)
-----------------------------------------------------------
 Add-on version: 2.9.1-2
 You are running the latest version of this add-on.
 System: Debian GNU/Linux 10 (buster)  (amd64 / qemux86-64)
 Home Assistant Core: 2021.12.9
 Home Assistant Supervisor: 2021.12.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums
 https://github.com/alexbelgium/hassio-addons
-----------------------------------------------------------
/scripts/99-run.sh: executing
/usr/sbin/inadyn: unrecognized option: forced-update
Usage:
 inadyn [-1hnsvC] [-c CMD] [-e CMD] [-f FILE] [-i IFNAME] [-I NAME] [-l LVL] [-p USR:GRP] [-P FILE] [-t SEC]
 -1, --once                     Run only once, updates if too old or unknown
     --force                    Force update, even if address has not changed
     --cache-dir=PATH           Persistent cache dir of IP sent to providers.
                                Default use ident NAME: /root/.cache/inadyn/
 -c, --cmd=/path/to/cmd         Script or command to run to check IP
 -C, --continue-on-error        Ignore errors from DDNS provider
 -e, --exec=/path/to/cmd        Script to run on DDNS update
     --exec-mode=MODE           Set script run mode: compat, event:
                                - compat: successful DDNS update only, default
                                - event: any update status
     --check-config             Verify syntax of configuration file and exit
 -f, --config=FILE              Use FILE name for configuration, default uses
                                ident NAME: /etc/inadyn.conf
 -h, --help                     Show summary of command line options and exit
 -i, --iface=IFNAME             Check IP of IFNAME instead of external server
 -I, --ident=NAME               Identity for config file, PID file, cache dir,
                                and syslog messages.  Defaults to: inadyn
 -l, --loglevel=LEVEL           Set log level: none, err, info, notice*, debug
 -n, --foreground               Run in foreground with logging to stdout/stderr
 -p, --drop-privs=USER[:GROUP]  Drop privileges after start to USER:GROUP
     --no-pidfile               Do not create PID file, for use with systemd
 -P, --pidfile=FILE             File to store process ID for signaling inadyn
                                Default uses ident NAME: /var/run/inadyn.pid
 -s, --syslog                   Log to syslog, default unless --foreground
 -t, --startup-delay=SEC        Initial startup delay, default none
 -v, --version                  Show program version and exit
Bug report address: https://github.com/troglobit/inadyn/issues
Project homepage: https://troglobit.com/projects/inadyn/
```

**Comment by alexbelgium** [comment]
indeed it didn't work :-) I'll try --force but still there is no timings... would setting the automation as I described above be a solution for you?

**Comment by alexbelgium** [comment]
New version pushed with --force

**Comment by X1pheR** [comment]
The version 2.9.1-3 works now! I didn't think of the automation, good suggestion. I'll do that as it is a good solution. Thanks!

**Comment by alexbelgium** [comment]
Perfect! If you are happy with that I'll close the comment, as I don't see an easy way to solve it and it seems perhaps due to the upstream app
